vm/systemSmartContracts/esdt.go

```diff
@@ -795,6 +795,11 @@ func (e *esdt) EpochConfirmed(epoch uint32) {
 	log.Debug("esdt contract", "enabled", e.flagEnabled.IsSet())
 }
 
+// SetNewGasCosts is called whenever a gas cost was changed
+func (e *esdt) SetNewGasCosts(gasCost vm.GasCost) {
+	e.gasCost = gasCost
```

Comment:
We need to protect `s.gasCost` with a mutex

Reply:
no need. there is no possibility to call the from different go routines

Reply:
protected all the way